src/Adapter/Presenter/Cart/CartPresenter.php

```diff
@@ -557,7 +558,22 @@ private function getTemplateVarVouchers(Cart $cart)
                 $freeShippingOnly = true;
             }
             if ($this->cartVoucherHasPercentReduction($cartVoucher)) {
-                $productsTotalExcludingTax = $cart->getOrderTotal($this->includeTaxes(), Cart::ONLY_PRODUCTS);
+                $products = $cart->getProducts();
+                if ($this->cartVoucherHasPercentReductionOnSelectedProducts($cartVoucher)) {
+                    $selected_products = $cartVoucher['obj']->checkProductRestrictionsFromCart($cart, true);
+                    if (is_array($selected_products)) {
+                        foreach ($products as $key => $product) {
+                            if ((in_array($product['id_product'] . '-' . $product['id_product_attribute'], $selected_products)
+                                    || in_array($product['id_product'] . '-0', $selected_products))
+                                && (($cartVoucher['reduction_exclude_special'] && !$product['reduction_applies'])
+                                    || !$cartVoucher['reduction_exclude_special'])) {
```

Comment:
Maybe this is worth adding a comment explaining this very complex condition ?

Reply:
Where is this complex :trollface: ?

Reply:
Comment added ;)